digitaltwin_client/inc/digitaltwin_interface_client.h

```diff
@@ -204,21 +204,31 @@ typedef void(*DIGITALTWIN_REPORTED_PROPERTY_UPDATED_CALLBACK)(DIGITALTWIN_CLIENT
 */
 typedef void(*DIGITALTWIN_CLIENT_TELEMETRY_CONFIRMATION_CALLBACK)(DIGITALTWIN_CLIENT_RESULT dtTelemetryStatus, void* userContextCallback);
 
+/** 
+  @brief Function signature for the callback that is invoked by the Digital Twin SDK when updating command's status asynchronously succeeds or fails.
+
+  @remarks When a device application invokes <c>DigitalTwin_InterfaceClient_UpdateAsyncCommandStatusAsync</c>, it may optionally include a function pointer callback with this signature.
+
+  @remarks When the Digital Twin either successfully sends the command status update to the service or fails, it will invoke this callback.
+
+  @param[in] dtUpdateAsyncCommandStatus         Status of the update operation.
+  @param[in] userContextCallback                Context pointer that was specified in the parameter <c>userContextCallback</c> during the device application's <c>DigitalTwin_InterfaceClient_UpdateAsyncCommandStatusAsync</c> call.
+
+  @return N/A
+*/
+typedef void(*DIGITALTWIN_CLIENT_UPDATE_ASYNC_COMMAND_CALLBACK)(DIGITALTWIN_CLIENT_RESULT dtUpdateAsyncCommandStatus, void* userContextCallback);
+
+
 /** 
   @brief Creates a <c>DIGITALTWIN_INTERFACE_CLIENT_HANDLE</c>
 
   @remarks The first step in setting up an interface is to create the underlying handle for it, using <c>DIGITALTWIN_INTERFACE_CLIENT_HANDLE</c> returned in <c>dtInterfaceClientHandle</c>.
            This handle cannot be used until it is registered with the service.
 
-  @remarks Note that unlike many SDK functions, where the <c>userInterfaceContext</c> would be tied to a specific operation and then never referenced again, in 
-           this case <c>userInterfaceContext</c> used throughout the life cycle of the interface handle.  In particular it is used when the interface's registration
-           status changes.  It is also used when commands and properties are invoked on this interface, assuming the interface accepts such callbacks.  This allows
-           the context to store state for the device application that spans the life of the interface and is not tied to a single call.
-
   @param[in]  interfaceId                       The interfaceId of interface to be registered.  For example, urn:contoso:com:EnvironmentalSensor:1.
   @param[in]  interfaceInstanceName             The interface instance name associated with this interface Id.  For example, environmentalSensor.
   @param[in]  dtInterfaceRegisteredCallback     (Optional) Function pointer to be invoked when the interface status (e.g. is registered or fails to register) changes.
-  @param[in]  userInterfaceContext              (Optional) Context pointer passed to dtInterfaceRegisteredCallback function on interface change, command updates, and property updates.
+  @param[in]  userInterfaceContext              (Optional) Context pointer passed to dtInterfaceRegisteredCallback function on interface changes.
```

Comment:
>userInterfaceContext [](start = 14, length = 20)

Nit: can we keep this consistent with other callbacks by naming it userContextCallback

Reply:
I need to sleep on this one and will check code in as it is for now.  It's a good point, but my thinking had been this is quite different than other cb's so was trying to name specify.  But "user interface" business could be problems.

---
In reply to: [322951649](https://github.com/Azure/azure-iot-sdk-c/pull/1169#discussion_r322951649) [](ancestors = 322951649)